pkgs/by-name/am/amber-lang/bash-path.patch

```diff
@@ -0,0 +1,146 @@
++                    );
+                 }
+                 Ok(tokens)
+-            },
++            }
+             Err((err_type, pos)) => {
+                 let error_message = match err_type {
+                     LexerErrorType::Singleline => {
+                         format!("Singleline {} not closed", pos.data.as_ref().unwrap())
+-                    },
++                    }
+                     LexerErrorType::Unclosed => {
+                         format!("Unclosed {}", pos.data.as_ref().unwrap())
+                     }
+@@ -81,18 +86,25 @@ impl AmberCompiler {
+         };
+         if Self::env_flag_set(AMBER_DEBUG_TIME) {
+             let pathname = self.path.clone().unwrap_or(String::from("unknown"));
+-            println!("[{}]\tin\t{}ms\t{pathname}", "Parsed".blue(), time.elapsed().as_millis());
++            println!(
++                "[{}]\tin\t{}ms\t{pathname}",
++                "Parsed".blue(),
++                time.elapsed().as_millis()
++            );
+         }
+         // Return result
+         match result {
+             Ok(()) => Ok((block, meta)),
+-            Err(failure) => Err(failure.unwrap_loud())
++            Err(failure) => Err(failure.unwrap_loud()),
+         }
+     }
+ 
+     pub fn translate(&self, block: Block, meta: ParserMetadata) -> String {
+         let imports_sorted = meta.import_cache.topological_sort();
+-        let imports_blocks = meta.import_cache.files.iter()
++        let imports_blocks = meta
++            .import_cache
++            .files
++            .iter()
+             .map(|file| file.metadata.as_ref().map(|meta| meta.block.clone()))
+             .collect::<Vec<Option<Block>>>();
+         let mut meta = TranslateMetadata::new(meta);
+@@ -105,7 +117,11 @@ impl AmberCompiler {
+         }
+         if Self::env_flag_set(AMBER_DEBUG_TIME) {
+             let pathname = self.path.clone().unwrap_or(String::from("unknown"));
+-            println!("[{}]\tin\t{}ms\t{pathname}", "Translate".magenta(), time.elapsed().as_millis());
++            println!(
++                "[{}]\tin\t{}ms\t{pathname}",
++                "Translate".magenta(),
++                time.elapsed().as_millis()
++            );
+         }
+         result.push(block.translate(&mut meta));
+         result.join("\n")
+@@ -119,22 +135,29 @@ impl AmberCompiler {
+ 
+     pub fn execute(code: String, flags: &[String]) {
+         let code = format!("set -- {};\n\n{}", flags.join(" "), code);
+-        Command::new("/bin/bash").arg("-c").arg(code).spawn().unwrap().wait().unwrap();
++        Command::new("/usr/bin/env")
++            .arg("bash")
++            .arg("-c")
++            .arg(code)
++            .spawn()
++            .unwrap()
++            .wait()
++            .unwrap();
+     }
+ 
+     #[allow(dead_code)]
+     pub fn test_eval(&mut self) -> Result<String, Message> {
+         self.compile().map_or_else(Err, |(_, code)| {
+             let child = Command::new("/bin/bash")
```

Comment:
It would make sense to change to me since `test_eval()` is what's being used for the tests -> https://github.com/Ph0enixKM/Amber/blob/4d4db1e384343d49e347e95fdd73e007f4c12fef/src/tests/validity.rs#L6

Reply:
I removed this patch in the latest push